SmokeTester/Program.cs

```diff
@@ -111,5 +113,20 @@ private static void WriteSummary(IReadOnlyDictionary<Uri, CrawledUrlProperties>
                 }
             }
         }
+
+        private static void WriteReferrers(IReadOnlyDictionary<Uri, CrawledUrlProperties> result, CrawlError error)
+        {
+            var referrersCount = result[error.Url].Referrers.Count();
```

Comment:
Are we `NullReferenceException` safe here?

Reply:
`CrawledUrlProperties.Referrers` is never `null`. `result[]` is an indexer on `IReadonlyDictionary` that will throw `KeyNotFoundException` rather than return `null`. Also, the access path has not changed compared to old code, so I think we're good here.